src/MOI_wrapper.jl

```diff
@@ -0,0 +1,355 @@
+using LinearAlgebra
+import MathOptInterface as MOI
+using ManifoldsBase
+using ManifoldDiff
+
+include("qp_block_data.jl")
+
+const _FUNCTIONS = Union{
+    MOI.VariableIndex,MOI.ScalarAffineFunction{Float64},MOI.ScalarQuadraticFunction{Float64}
+}
+
+"""
+    struct VectorizedManifold{M} <: MOI.AbstractVectorSet
+        manifold::M
+    end
+
+Representation of points of `manifold` as a vector of `R^n` where `n` is
+`MOI.dimension(VectorizedManifold(manifold))`.
+"""
+struct VectorizedManifold{M} <: MOI.AbstractVectorSet
+    manifold::M
+end
+
+function MOI.dimension(set::VectorizedManifold)
+    return prod(ManifoldsBase.representation_size(set.manifold))
+end
+
+struct _EmptyNLPEvaluator <: MOI.AbstractNLPEvaluator end
+MOI.features_available(::_EmptyNLPEvaluator) = [:Grad, :Jac, :Hess]
+MOI.initialize(::_EmptyNLPEvaluator, ::Any) = nothing
+
+"""
+    Manopt.Optimizer()
+
+Creates a new optimizer object.
+
+The minimization of a function `f(X)` of an an array `X[1:n1,1:n2,...]`
+over a manifold `M` starting at `X0`, can be modeled as follows:
+```julia
+using JuMP
+@variable(model, X[i1=1:n1,i2=1:n2,...] in M, start = X0[i1,i2,...])
+@objective(model, Min, f(X))
+```
```

Comment:
maybe add that this works for all manifolds that have an array representation by default.

Reply:
Done